stan/math/rev/core/build_vari_array.hpp

```diff
@@ -0,0 +1,23 @@
+#ifndef STAN_MATH_REV_CORE_BUILD_VARI_ARRAY_HPP
+#define STAN_MATH_REV_CORE_BUILD_VARI_ARRAY_HPP
+
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <stan/math/rev/core/var.hpp>
+#include <stan/math/rev/core/vari.hpp>
+
+namespace stan {
+namespace math {
+
+template <int R, int C>
+vari** build_vari_array(const Eigen::Matrix<var, R, C>& x) {
+  vari** x_vi_
+      = ChainableStack::instance().memalloc_.alloc_array<vari*>(x.size());
+  for (int i = 0; i < x.size(); ++i) {
+    x_vi_[i] = x.data()[i].vi_;
```

Comment:
x might actually be a matrix (and it's still a valid use case), so I think it makes sense to stick with x.data()[i]

Reply:
In which case, this should be just `x(i).vi_`.  The memory organization guarantee ensures this is the same as `x.data()[i]`.  But it will also generalize to more generic matrix types for `x` not that that matters.

The point isn't efficiency, it's to remove complexity.  Someone's going to see this and wonder why it needs the `.data()`.